### ISSUE TITLE
Add Parametrized Tests

### ISSUE THREAD

**Issue by kaosdev** [kaosdev/test-decorators#4]
## Description
Add a new decorator that will allow developers to write parametrized tests.
```typescript
@Parametrized({
   source: [1, 2, 3],
})
testSomething(data: number) {
   // The test will be executed 3 times with 'value' based on data
   ...
}
```
The test name should change adding an index at each iteration by default.
Optionally you can pass a custom name function.
```typescript
@Parametrized({
   source: [1, 2, 3],
   name: (data, index) => `${index}. testSomething(${data})`
})
```
Producing the following tests
```
0. testSomething(1)
1. testSomething(2)
2. testSomething(3)
```

## Why it is needed?
Sometimes you want to test the same thing but with just different input data.
A decorator will allow to write these tests in a clear syntax.

**Comment by kaosdev** [comment]
#5

**Comment by kaosdev** [comment]
Closed by #5
